python/across_api/base/fov.py

```diff
@@ -0,0 +1,382 @@
+                nside=uniq_nside[nonzero_prob_pixels],  # type: ignore
+                order="NESTED",
+            )
+        else:
+            ra, dec = ah.healpix_to_lonlat(
+                nonzero_prob_pixels, nside=nside, order=healpix_order
+            )
+
+        # Convert these coordinates into a SkyCoord
+        skycoord = SkyCoord(ra=ra, dec=dec, unit="deg")
+
+        # Calculate pixel indicies of the all the regions inside of the FOV
+        visible_probability_pixels = nonzero_prob_pixels[self.in_fov(skycoord=skycoord)]
+        # Calculate the amount of probability inside the FOV
+        if healpix_order == "NUNIQ":
+            # Calculate probability in FOV by multiplying the probability density by
+            # area of each pixel and summing up
+            pixarea = ah.nside_to_pixel_area(uniq_nside[visible_probability_pixels])
+            return float(
+                round(
+                    np.sum(healpix_loc[visible_probability_pixels] * pixarea.value), 5
+                )
+            )
+        else:
+            # Calculate the amount of probability inside the FOV
+            return float(round(np.sum(healpix_loc[visible_probability_pixels]), 5))
+
+
+class FootprintFOV(FOVBase):
+    """
+    Constrained instrumet FOV. This is an FOV that calculate lates what is
+    visible in the FOV at a given pointing
+    """
+
+    footprint: Footprint
+
+    def __init__(self, pointing: PointingBase) -> None:
+        projected_footprint = self.footprint.project(
+            ra=pointing.ra, dec=pointing.dec, pos_angle=pointing.position_angle
+        )
+
+        ras_poly = [x[0] for x in projected_footprint][:-1]
+        decs_poly = [x[1] for x in projected_footprint][:-1]
+
+        cartesian_vertices = spherical_to_cartesian(
+            1,  # radial component
+            np.deg2rad(decs_poly),  # LAT - DECs
+            np.deg2rad(ras_poly),  # LON - RAs
+        )
+
+        self.visible_pixels = hp.query_polygon(
+            hp.order2nside(HEALPIX_MAP_EVAL_ORDER),
+            np.array(cartesian_vertices).T,
+            inclusive=True,
+            nest=True,
+        )
+
+
+class AllSkyFOV(FOVBase):
+    """
+    All sky instrument FOV. This is a simple FOV that is always visible unless
+    Earth occulted.
+    """
+
+    earth_constraint: EarthLimbConstraint
+
+    def __init__(self, ephem: EphemBase, time: Time):
+        """
+        Finds all healpix pixels within the ephemeris time slot that are not earth
+        occulted, and sets them to FOVBase.visible_pixels.
+
+        Function does assume that it is a space mission with available ephemeris
+        """
+        n_side = hp.order2nside(HEALPIX_MAP_EVAL_ORDER)
+        n_pix = hp.order2npix(HEALPIX_MAP_EVAL_ORDER)
+        i_slice = get_slice(time=time, ephem=ephem)
+
+        ra = ephem.earth.ra[i_slice].degree
+        dec = ephem.earth.dec[i_slice].degree
+
+        vec = hp.ang2vec(ra, dec, lonlat=True)
```

Comment:
```suggestion
        vec = ephem.earth.cartesian.xyz.value
```

Reply:
I tried this, and it didn't give the anticipated results in the tests.